hack/lib/util.sh

```diff
@@ -496,10 +496,45 @@ kube::util::ensure_single_dir_gopath() {
   fi
 }
 
+# Checks whether there are any files matching pattern $2 changed between the
+# HEAD of current branch and the base ref, or upstream branch (if there's no PULL_BASE_SHA).
+# Returns 1 (false) if there are no changes
+#         0 (true)  if there are changes detected.
+kube::util::has_changes() {
+  local -r git_branch=$1
+  local -r pattern=$2
+  local -r not_pattern=${3:-totallyimpossiblepattern}
+
+  if [[ -n ${PULL_BASE_SHA:-} ]]; then
+    return $(has_changes_against_pull_base_sha pattern not_pattern)
+  fi
+
+  return $(has_changes_against_upstream_branch git_branch pattern not_pattern)
+}
+
+# Checks whether there are any files matching pattern $2 changed between the
+# HEAD of current branch and PULL_BASE_SHA.
+# Returns 1 (false) if there are no changes or PULL_BASE_SHA does not exist
```

Comment:
this currently returns 0 if `$PULL_BASE_SHA` isn't set?

Reply:
:+1: good catch